### PR TITLE
[GSSoC 26] AI-Powered Image Restoration and Enhancement Tool - Issue #1003Ai image restoration issue 1003

### LINKED ISSUE #1003: AI-Powered Image Restoration and Enhancement Tool
---
name: DL-Simplified Issue Template
about: This is the issue template that is going to be used while creating any issues
  in this repo.
title: ''
labels: ''
assignees: ''

---

### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : </br> AI-Powered Image Restoration and Enhancement Tool
:red_circle: **Aim** : </br> This project aims to develop an AI-Powered Image Restoration Tool that revitalizes low-quality historical photos by upscaling, denoising, and adding realistic color. 
:red_circle: **Dataset** : </br> Pretrained model
:red_circle: **Approach** :  Using advanced deep learning techniques, it transforms degraded images into vibrant, high-quality visuals while preserving their historical context. This tool is perfect for heritage conservation, family archiving, and historical documentation. 
SRCNN (Super-Resolution Convolutional Neural Network): For image upscaling
DnCNN: For image denoising
Pre-trained colorization models (e.g., U-Net): For adding color to grayscale images

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Venkata Ramana Billana
- GitHub Profile Link :  https://github.com/ramana2074
- Email ID : billanavenkataramana@gmail.com
- Participant ID (if applicable):
- Approach for this Project : Using advanced deep learning techniques, it transforms degraded images into vibrant, high-quality visuals while preserving their historical context. This tool is perfect for heritage conservation, family archiving, and historical documentation. 
- What is your participant role? (Mention the Open Source program) - SWOC

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

### PR DESCRIPTION
**Pull request by GoliJyothish**
# Pull Request for DL-Simplified 💡
## Issue Title : AI-Powered Image Restoration and Enhancement Tool
- **Info about the related issue (Aim of the project)** : Build an AI-powered pipeline for image restoration and enhancement using DnCNN, SRCNN, U-Net and ESRGAN models
- **Name:** Goli Jyothish
- **GitHub ID:** GollJyothish
- **Email ID:** jyothishgoli@email.com
- **Identify yourself:** GSSoC 2026 Participant

Closes: #1003

### Describe the add-ons or changes you've made 📃
Added a complete Jupyter notebook implementing 4 deep learning models for image restoration:
- DnCNN for image denoising
- SRCNN for super resolution
- U-Net for image colorization
- ESRGAN for enhanced super resolution (4x)
- Full evaluation using PSNR, SSIM and MSE metrics
- BSD300 dataset used for training and testing

## Type of change ☑️
- [ ] Bug fix
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update
- [ ] Breaking change
- [x] This change requires a documentation update

## How Has This Been Tested? ⚙️
- Tested on Google Colab with T4 GPU
- All 4 models run successfully on BSD300 dataset
- PSNR and SSIM metrics computed for each model
- Visual before/after comparisons generated for all models

## Checklist: ☑️
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
Please follow the project structure:
```
Project Folder Name/
|
├── Dataset/
|   ├── dataset.csv     <- Upload the dataset file inside the Dataset folder
|   └── README.md       <- Dataset details. If the dataset file is too large, put the link.
|
├── Images/
|   ├── img1.png        <- EDA images
|   ├── img2.png        <- Model related images
|   ├── img3.png        <- Accuracy score related images
|   └── img4.png        <- confusion matrix related images
|
├── Model/
|   ├── project_folder_name.ipynb  <- Jupyter notebook file (master)
|   └── README.md       <- README file on the Models
|
├── Web App/
|   ├── web_app.py      <- Web app to demo the project
|   ├── README.md       <- README file to demo and overview of the web app
|   └── web_app.mp4     <- Recorded video for the demo of the web app. To be included in README.
|
├── README.md           <- Master README.md file for the project folder
└── requirements.txt    <- requirements file for the list of plugin(s) & functions
```

Put your files according to the project structure mentioned and follow it properly. Without this project structure, the PRs will not be merged to this repository.

For README templates check the `.github` folder of the main repository: [Here it is](https://github.com/abhisheks008/DL-Simplified/tree/main/.github)

@GoliJyothish

**Comment by GoliJyothish** [comment]
Hi @abhisheks008! I have updated the project structure as requested. 

The folder now contains:
- Dataset/README.md ✅
- Images/README.md ✅
- Model/notebook + README.md ✅
- README.md ✅
- requirements.txt ✅

Please review and let me know if any further changes are needed.

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
1. Images folder should contain the project generated images/output. README file is good enough to tell the user what about this folder is, but the contents are required al so.
2. Please follow the README.md template and update the same (For Main README): https://github.com/abhisheks008/DL-Simplified/tree/main/.github

@GoliJyothish

**Comment by GoliJyothish** [comment]
Hi @abhisheks008! I have addressed both the requested changes:

1. ✅ **Images/README.md** — Updated to describe all output images generated from the notebook (EDA images, model comparison images, PSNR/SSIM visualizations)

2. ✅ **Main README.md** — Updated to follow the repository README template with all required sections:
   - Goal
   - Dataset
   - Description
   - What I had done
   - Models Implemented
   - Libraries Needed
   - EDA Results
   - Performance & Accuracy Scores
   - Conclusion
   - Signature

Please review and let me know if any further changes are needed.

**Comment by GoliJyothish** [comment]
Hi @abhisheks008! Just following up on this PR since it's been a few weeks. 
I've addressed both requested changes (Images/README.md and Main README.md 
following the template). Please let me know if anything else needs updating. 
Thank you for your time!

**Review by abhisheks008 — APPROVED** [review]
Approved
